needle.pm

```diff
@@ -79,7 +81,7 @@ sub new {
         return;
     }
 
-    $self->{file} = $jsonfile;
+    $self->{file} = substr($jsonfile, length($bmwqemu::vars{PRJDIR}) + 1);
```

Comment:
this looks very dangerous. I would prefer if you split prefix directory and relative filename somewhere above and die then if the prefix directory doesn't match - and take the relative filename here. The 2 statements are 30 lines apart - which makes them dangerous. One cache refactoring later and we're screwed